api/src/main/java/com/ford/labs/retroquest/metrics/Metrics.java

```diff
@@ -25,54 +19,59 @@ public Metrics(TeamRepository teamRepository, FeedbackRepository feedbackReposit
 
     @ManagedAttribute
     public int getTeamCount() {
-        return teamRepository.findAll().size();
+        return getTeamCount(null, null);
     }
 
-    public long getTeamCount(LocalDate startDate, LocalDate endDate) {
-        if (startDate == null && endDate == null) {
-            return teamRepository.count();
-        } else if (endDate == null) {
-            return teamRepository.countAllByDateCreatedAfterAndDateCreatedIsNotNull(startDate);
-        }
-        return teamRepository.countAllByDateCreatedBetweenAndDateCreatedNotNull(startDate, endDate);
+    public int getTeamCount(LocalDate startDate, LocalDate endDate) {
+        var dateRange = DateTimeRange.fromStartAndEnd(startDate, endDate);
+
+        return (int) teamRepository
+            .countAllByDateCreatedBetweenAndDateCreatedNotNull(
+                dateRange.getStartDate(),
+                dateRange.getEndDate()
+            );
     }
 
     @ManagedAttribute
     public int getFeedbackCount() {
-        return feedbackRepository.findAll().size();
+        return getFeedbackCount(null, null);
     }
 
-    public int getFeedbackCount(LocalDate startTime, LocalDate endTime) {
-        LocalDateTime finalEndTime = endTime == null ? LocalDateTime.now() : endTime.atStartOfDay();
-        LocalDateTime finalStartTime = startTime == null ? LocalDateTime.MIN : startTime.atStartOfDay();
-        List<Feedback> feedbackList = feedbackRepository.findAll().stream()
-                .filter(feedback -> !feedback.getDateCreated().isBefore(finalStartTime))
-                .filter(feedback -> !feedback.getDateCreated().isAfter(finalEndTime))
-                .collect(toList());
-        return feedbackList.size();
+    public int getFeedbackCount(LocalDate startDate, LocalDate endDate) {
+        var dateRange = DateTimeRange.fromStartAndEnd(startDate, endDate);
+
+        return (int) feedbackRepository
```

Comment:
primnitive obsession

Usage of primitive data type. Also  will there be any issue if we could send Long as it is?

The purpose of keeping the return data type is Long is kept in JPA to allow the maximum count.

Reply:
The int is used in order to provide metrics to JMX. I'm not sure if we want to keep the JMX implementation around long-term, so if we end up removing that we'll make sure to change all these return types to `long` then.

Reply:
Also instead of passing these getStartDateTime  getEndDateTime again here 

pass DateTimeRange to the repository by having a seperate query in repo something similar like

```

   @Query("select avg(f.stars) from Feedback f where f.stars > 0 and f.dateCreated >= :#{#dateTimeRange.startDateTime} and f.dateCreated <= :#{#dateTimeRange.endDateTime}")
    double getAverageRating(@Param("dateTimeRange") DateTimeRange dateTimeRange);

```


I have done for feedback repo , you can create similar for this query as well